src/ddprof_cmdline.cc

```diff
@@ -135,19 +135,24 @@ static bool watcher_from_config(EventConf *conf, PerfWatcher *watcher) {
 }
 
 // If this returns false, then the passed watcher should be regarded as invalid
-bool watchers_from_str(const char *str, std::vector<PerfWatcher> &watchers) {
+bool watchers_from_str(const char *str, std::vector<PerfWatcher> &watchers,
+                       uint32_t sample_stack_user) {
   std::vector<EventConf> configs;
   if (EventConf_parse(str, configs) != 0) {
     return false;
   }
 
   for (auto &conf : configs) {
+    // todo{@r1viollet}: ideally here we would want to change the template
+    // value of the event conf. Instead we override the conf value.
+    if (conf.sample_stack_user == k_default_perf_sample_stack_user) {
```

Comment:
Logic is strange, what happens if I do `ddprof --sample-stack-user 65528 -e sCPU,s=32768` ?

Reply:
`s=32768` wins: conf will hold the value from the watcher configuration, so global parameter is ignored. I'm not a fan of this. Alternative ideas are welcome.

Reply:
I would indeed expect that event parameter takes precedence, but that's not what is written in this function.
I believe that with the current code, event will have its sample stack size set to 65528.

Reply:
Indeed, I did not account for the case where user would re-force the value to the default.

Reply:
I fixed this implementing the todo.